cmake/thirdparty/get_legion.cmake

```diff
@@ -70,6 +70,14 @@ function(find_or_configure_legion)
 
   if(Legion_FOUND)
     message(STATUS "CPM: using local package Legion@${version}")
+
+    # Check that required Legion options are set
+    if(NOT Legion_USE_Python)
+      message(FATAL_ERROR "Legion was not compiled with Legion_USE_Python")
+    endif()
```

Comment:
This makes it impossible to build the legate C++ library without Legion built w/ Python bindings. Is this really what we want?

Reply:
This change is my attempt at avoiding the following fell-bad scenario:

* user builds Legion separately, forgets to add python support
* user builds legate.core C++ bits using `cmake` directly
* users does `pip install`
* user realizes nothing is working, because they built Legion w/o Python support (and the build process never complained)

I would also be ok with a situation where the cmake build doesn't enforce `Legion_USE_Python`, and instead that becomes necessary only when you try to install the python pieces of legate.core. Do you think that's possible/preferable? Or maybe you have a different approach to avoid this scenario?